app/models/user.rb

```diff
@@ -963,17 +963,13 @@ def ensure_uniq_email
     raise Exceptions::UnprocessableEntity, 'Email address is already used for other user.'
   end
 
-  def validate_roles
-    return true if !role_ids
-    role_ids.each do |role_id|
-      role = Role.lookup(id: role_id)
-      raise "Unable to find role for id #{role_id}" if !role
-      next if !role.preferences[:not]
-      role.preferences[:not].each do |local_role_name|
-        local_role = Role.lookup(name: local_role_name)
-        next if !local_role
-        raise "Role #{role.name} conflicts with #{local_role.name}" if role_ids.include?(local_role.id)
-      end
+  def validate_roles(role)
+    return true if !role_ids # we need role_ids for checking in role_ids below, in this method
+    return true if role.preferences[:not].blank?
+    role.preferences[:not].each do |local_role_name|
+      local_role = Role.lookup(name: local_role_name)
+      next if role_ids.exclude?(local_role.id)
```

Comment:
Somehow the line `next if !local_role` got lost. This leads to a potential bug where the configured `:not` `local_role_name` is not valid and `Role.lookup(name: local_role_name)` returns `nil`. If so accessing `local_role.id` will fail hard. Please reintroduce the guard clause for this case.

Reply:
yeah you are right. i missed that part if `local_role` is nil. Will add the guard clause.